clients/go/cacclient/main.go

```diff
@@ -0,0 +1,97 @@
+package cacclient
+
+/*
+#include "libcac_client.h"
+*/
+import "C"
+import (
+	"fmt"
+	"unsafe"
+)
+
+// CacClient struct
+type CacClient struct {
+	tenant           string
+	pollingFrequency int
+	cacHostName      string
+}
+
+// NewCacClient creates a new CacClient
+func NewCacClient(tenantName string, pollingFrequency int, cacHostName string) *CacClient {
+	return &CacClient{tenant: tenantName, pollingFrequency: pollingFrequency, cacHostName: cacHostName}
+}
+
+// GetCacLastErrorMessage gets the last error message from the C library
+func (c *CacClient) GetCacLastErrorMessage() string {
+	return C.GoString(C.cac_last_error_message())
+}
+
+// GetCacLastErrorLength gets the length of the last error message
+func (c *CacClient) GetCacLastErrorLength() int {
+	return int(C.cac_last_error_length())
+}
+
+// GetCacClient gets the client pointer from the C library
+func (c *CacClient) GetCacClient() *C.Arc_Client {
+	return C.cac_get_client(C.CString(c.tenant))
+}
+
+// CreateNewCacClient creates a new client in the C library
+func (c *CacClient) CreateNewCacClient() int {
+	resp := C.cac_new_client(C.CString(c.tenant), C.ulong(c.pollingFrequency), C.CString(c.cacHostName))
+	if resp == 1 {
+		errorMessage := c.GetCacLastErrorMessage()
+		fmt.Printf("Some Error Occur while creating new client: %s\n", errorMessage)
+	}
+	return int(resp)
+}
+
+// StartCacPollingUpdate starts polling updates in a separate goroutine
+func (c *CacClient) StartCacPollingUpdate() {
+	tenant := C.CString(c.tenant)
+	go C.cac_start_polling_update(tenant)
+}
+
+// GetCacConfig gets the configuration from the C library
+func (c *CacClient) GetCacConfig(filterQuery string, filterPrefix string) string {
```

Comment:
where ever we are accepting filters, they have to go equivalent of `Option<Vec<String>>` and then we get Some value then we join the array based on the join character for that particular function, else we send null pointer

Reply:
done